src/Mvc/Mvc.Core/src/Builder/OpenApiEndpointConventionBuilderExtensions.cs

```diff
@@ -122,36 +123,85 @@ public static DelegateEndpointConventionBuilder ExcludeFromDescription(this Dele
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds <see cref="IAcceptsMetadata"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="DelegateEndpointConventionBuilder"/>.</param>
+        /// <param name="requestType">The type of the request body.</param>
+        /// <param name="contentType">The request content type that the endpoint accepts.</param>
+        /// <param name="additionalContentTypes">Additional request content types that the endpoint accepts.</param>
         /// <returns>A <see cref="DelegateEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
         public static DelegateEndpointConventionBuilder Accepts(this DelegateEndpointConventionBuilder builder,
             Type requestType, string contentType, params string[] additionalContentTypes)
         {
-            builder.WithMetadata(new ConsumesAttribute(requestType, contentType, additionalContentTypes));
+            builder.WithMetadata(new AcceptsMetadata(requestType, true, GetAllContentTypes(contentType, additionalContentTypes)));
+            return builder;
+        }
+
+
+        /// <summary>
+        /// Adds <see cref="IAcceptsMetadata"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="DelegateEndpointConventionBuilder"/>.</param>
+        /// <param name="requestType">The type of the request body.</param>
+        /// <param name="isOptional">Indicates whether the request body is required or not.</param>
+        /// <param name="contentType">The request content type that the endpoint accepts.</param>
+        /// <param name="additionalContentTypes">Additional request content types that the endpoint accepts.</param>
+        /// <returns>A <see cref="DelegateEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static DelegateEndpointConventionBuilder Accepts(this DelegateEndpointConventionBuilder builder,
+            Type requestType, bool isOptional, string contentType, params string[] additionalContentTypes)
+        {
+            builder.WithMetadata(new AcceptsMetadata(requestType, isOptional, GetAllContentTypes(contentType, additionalContentTypes)));
             return builder;
         }
+
+        private static string[] GetAllContentTypes(string contentType, string[] additionalContentTypes)
+        {
+            var allContentTypes = new string[additionalContentTypes.Length + 1];
+            allContentTypes[0] = contentType;
+
+            for (var i = 0; i < additionalContentTypes.Length; i++)
+            {
+                allContentTypes[i + 1] = additionalContentTypes[i];
+            }
```

Comment:
Super unnecessary, but you can do this:

```suggestion
            additionalContentTypes.CopyTo(allContentTypes.AsSpan(1));
```

Reply:
Any perf impact ?

Reply:
I'm guessing the for loop is probably faster, but this lets you save trees if you're like me and print out all PRs to review.